### PR TITLE
Adding Egyptian Domestic Terrorist List crawler

### PR DESCRIPTION
**Pull request by bgmello**
Fixes opensanctions/crawler-planning#124

### CODE REVIEW

**Comment by bgmello** [comment]
It's still warning of an invalid nationality even though there is the lookup for that string.

File: datasets/eg/terrorists/crawler.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+def crawl_terrorist(input_dict: dict, context: Context):
+
+    first_name = input_dict.pop("alasm")
+    last_name = input_dict.pop("asm_akhr")
+
+    person = context.make("Person")
+    person.id = context.make_id(first_name, last_name)
+    h.apply_name(person, first_name=first_name, last_name=last_name)
+    person.add("nationality", input_dict.pop("aljnsyt"), lang="ar")
```
[the remark this answers is not in the document]
Comment (jbothma) [reply datasets/eg/terrorists/crawler.py:86]: e.g. https://github.com/opensanctions/opensanctions/blob/main/datasets/id/dttot/id_dttot.yml#L34 with something like

```python
translated_row = {}
for key, value in row.items():
    translated_key = context.lookup_value(key)
    translated_row[translated_key] = value
yield translated_row
```

File: datasets/eg/terrorists/crawler.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            record[header] = stringify(value)
+        if len(record) == 0:
+            continue
+        yield record
+
+
+def crawl_terrorist(input_dict: dict, context: Context):
+
+    first_name = input_dict.pop("alasm")
+    last_name = input_dict.pop("asm_akhr")
```
[the remark this answers is not in the document]
Comment (jbothma) [reply datasets/eg/terrorists/crawler.py:81]: Google Translate gives me 
image
so I think this should more be `name` and `alias` than `firstName` and `lastName`

File: datasets/eg/terrorists/crawler.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    if not name:
+        return
+
+    entity = context.make("LegalEntity")
+    entity.id = context.make_id(name)
+    entity.add("name", name)
+
+    sanction = h.make_sanction(context, entity)
+    sanction.add("recordId", input_dict.pop("rqm_alqdyt"))
+    sanction.add("summary", input_dict.pop("althdythat"), lang="ar")
```
Comment (jbothma) [inline datasets/eg/terrorists/crawler.py:120]: could you make all the langs 3-letter?

```suggestion
    sanction.add("summary", input_dict.pop("althdythat"), lang="ara")
```

File: datasets/eg/terrorists/crawler.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+
+def crawl_legal_persons(input_dict: dict, context: Context):
+
+    name = input_dict.pop("asm_alshkhs_alatbary")
+
+    if not name:
+        return
+
+    company = context.make("Company")
```
Comment (jbothma) [inline datasets/eg/terrorists/crawler.py:134]: As the sheet name suggests, this is a bit broader than just companies. I see one charity

```suggestion
def crawl_legal_persons(input_dict: dict, context: Context):

    name = input_dict.pop("asm_alshkhs_alatbary")

    if not name:
        return

    company = context.make("LegalEntity")
```

File: datasets/eg/terrorists/crawler.py [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+def crawl_terrorist_entities(input_dict: dict, context: Context):
+
+    name = input_dict.pop("asm_alkyan")
+
+    if not name:
+        return
+
+    entity = context.make("LegalEntity")
+    entity.id = context.make_id(name)
+    entity.add("name", name)
```
Comment (jbothma) [inline datasets/eg/terrorists/crawler.py:116]: let's add case number (رقم القضية) to ID for all types/sheets - no need to parse - just as i is.

**Review by jbothma — COMMENTED** [review]
I don't see a nationality warning, but I do see linting warnings :)

File: datasets/eg/terrorists/crawler.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jbothma) [reply datasets/eg/terrorists/crawler.py]: thanks, this is way more readable!

**Pull request merged**